docker-compose.yml

```diff
@@ -0,0 +1,142 @@
+version: '3.1'
+volumes:
+  data:
+  capnp-secrets:
+  worker-data:
+  scheduler-data:
+  caddy_data:
+  caddy_config:
+
+secrets:
+  ocaml-ci-github-key:
+    file: $HOME/ci_secrets/new-test-ci.2022-11-08.private-key.pem
+  ocaml-ci-webhook-secret:
+    file: $HOME/ci_secrets/benmandrew-ci-webhook-secret
+
+services:
+  caddy:
+    image: caddy
+    ports:
+      - 80:80
+      - 443:443
+      - 8100:8100
+    volumes:
+      - /etc/caddy:/etc/caddy:ro
+      - caddy_data:/data
+      - caddy_config:/config
+
+  service:
+    image: ocaml-ci-service
+    # For local deploys using docker build -t ocaml-ci-service -f Dockerfile .
+    command: >
+      --github-app-id 244548
+      --github-private-key-file /run/secrets/ocaml-ci-github-key
+      --github-webhook-secret-file /run/secrets/ocaml-ci-webhook-secret
+      --confirm above-average
+      --confirm-auto-release 120
+      --capnp-public-address=tcp:service:9000
+      --capnp-listen-address=tcp:0.0.0.0:9000
+      --submission-service /capnp-secrets/user.cap
+      --submission-solver-service /capnp-secrets/user.cap
+      --migration-path /migrations
+      --github-account-allowlist 'benmandrew'
+    depends_on:
+      - scheduler
+    restart: on-failure
+    environment:
+      - "CAPNP_PROFILE=production"
+      - "PLATFORMS=minimal"
+      - "DOCKER_BUILDKIT=1"
+      - "PROGRESS_NO_TRUNC=1"
+    ports:
+      - ':9000'
```

Comment:
Ah I misunderstood what the `ports` configuration does, I'll remove it.

There is no host port necessary for `service`, as caddy exposes a host port and provides access to `service` by reverse proxy. It allows you do access `service` by `localhost:8100` rather than having to know the IP of the container.